bin/release.js

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+import {fileURLToPath} from 'url'
+import {dirname, join} from 'pathe'
+import {createRequire} from 'module'
+import {readFileSync} from 'fs'
+import {temporaryDirectoryTask} from 'tempy'
+
+const require = createRequire(import.meta.url)
+const {Octokit} = require('@octokit/rest')
+const execa = require('execa')
+
+const octokit = new Octokit({
+    auth: process.env.GITHUB_TOKEN,
+});
+const supportedOSs = ['linux', 'windows', 'darwin']
+const supportedArchitectures =  ['386', 'amd64']
+const goPath = process.env.PATHGO
+const baseReleaseParams = {
+    owner: 'shopify',
+    repo: 'cli',
+}
+const binDirectory = dirname(fileURLToPath(import.meta.url))
+const rootDirectory = dirname(binDirectory)
+const extensionsPath = join(rootDirectory, 'packages/ui-extensions-go-cli')
+let cliVersion = getCliVersion()
+
+const versionsMatrix = generateVersionsMatrix()
+temporaryDirectoryTask(async (directory) => {
+    await Promise.all(versionsMatrix.map((version) => packageVersion(version, directory)))
+    const release = await createRelease()
+    await Promise.all(versionsMatrix.map((version) => updloadReleaseAssetVersion(version, release, directory)))
+})
+
+function generateVersionsMatrix() {
+    return supportedOSs.flatMap((os) => supportedArchitectures.map((arch) => {
+        return {
+            os,
+            arch: (os === 'darwin' && arch === '386') ? 'arm64' : arch
+        }
+    }))
+}
```

Comment:
Maybe you are right and an array of [OS, ARCH] will do the work without adding complex instructions. We could modify in a future PR in case people finds it a bit cumbersome